src/fu-util.c

```diff
@@ -463,6 +463,179 @@ fu_util_perhaps_show_unreported (FuUtilPrivate *priv, GError **error)
+		return NULL;
+	}
+	file = g_file_new_for_path (as_format_get_filename (format));
+	root = as_node_from_file (file, AS_NODE_FROM_XML_FLAG_NONE, NULL, error);
+	if (root == NULL)
+		return NULL;
+
+	/* manually find the first agreement section */
+	n = as_node_find (root, "component/agreement/agreement_section/description");
+	if (n == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_FOUND,
+				     "No agreement description found");
+		return NULL;
+	}
+	str = as_node_to_xml (n->children, AS_NODE_TO_XML_FLAG_INCLUDE_SIBLINGS);
+	tmp = as_markup_convert_simple (str->str, error);
+#endif
+	return g_string_new (tmp);
+}
+
+static gboolean
+fu_util_modify_remote_warning (FuUtilPrivate *priv, FwupdRemote *remote, GError **error)
+{
+	AsApp *app;
+	g_autofree gchar *component_id = NULL;
+	g_autoptr(AsStore) store = NULL;
+	g_autoptr(GHashTable) os_release = NULL;
+	g_autoptr(GString) desc_plain_str = NULL;
+
+	/* try to find a custom agreement, falling back to a generic warning */
+	store = as_store_new ();
+	as_store_add_filter (store, AS_APP_KIND_SOURCE);
+	if (!as_store_load (store, AS_STORE_LOAD_FLAG_APPDATA, NULL, error))
+		return FALSE;
+	component_id = g_strdup_printf ("org.freedesktop.fwupd.remotes.%s",
+					fwupd_remote_get_id (remote));
+	app = as_store_get_app_by_id (store, component_id);
+	if (app != NULL) {
+		desc_plain_str = fu_util_get_component_agreement (priv, app, error);
+		if (desc_plain_str == NULL)
+			return FALSE;
+	} else {
+		desc_plain_str = fu_util_get_remote_warning (priv, remote, error);
+		if (desc_plain_str == NULL)
+			return FALSE;
+	}
+
+	/* replace any dynamic values from os-release */
+	os_release = fwupd_get_os_release (error);
+	as_utils_string_replace (desc_plain_str, "$OS_RELEASE:NAME$",
+				 g_hash_table_lookup (os_release, "NAME"));
+	as_utils_string_replace (desc_plain_str, "$OS_RELEASE:BUG_REPORT_URL$",
+				 g_hash_table_lookup (os_release, "BUG_REPORT_URL"));
+
+	/* show and ask user to confirm */
+	g_print ("%s", desc_plain_str->str);
+	if (!priv->assume_yes) {
+		/* ask for permission */
+		g_print ("\n%s [Y|n]: ",
+			 /* TRANSLATORS: should the remote still be enabled */
+			 _("Agree and enable the remote?"));
+		if (!fu_util_prompt_for_boolean (TRUE)) {
+			g_set_error_literal (error,
+					     FWUPD_ERROR,
+					     FWUPD_ERROR_NOTHING_TO_DO,
+					     "Declined agreement");
+			return FALSE;
+		}
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_util_modify_remote (FuUtilPrivate *priv,
+		       const gchar *remote_id,
+		       const gchar *key,
+		       const gchar *value,
+		       GError **error)
+{
+	g_autoptr(FwupdRemote) remote = NULL;
+
+	/* ensure the remote exists */
+	remote = fwupd_client_get_remote_by_id (priv->client, remote_id, NULL, error);
+	if (remote == NULL)
+		return FALSE;
+
+	/* show some kind of warning when enabling */
+	if (g_strcmp0 (key, "Enabled") == 0 && g_strcmp0 (value, "true") == 0) {
```

Comment:
After looking at this I noticed something that is worth following up in another commit.  **modify-remote** output shows "Enabled: True" or "Enabled: False" but that gives the false presumption that you send "True" or "False" to it to enable/disable where you actually want to send "true" or "false".

Reply:
Fixed, thanks.